### ISSUE TITLE
Invalid HTML for select element

### ISSUE THREAD

**Issue by nlhkabu** [pypa/warehouse#6414]
This html is generated by the Python form code.

template:
https://github.com/pypa/warehouse/blob/master/warehouse/templates/manage/roles.html

field:
`{{ form.role_name }}`

ERROR: The first child “option” element of a “select” element with a “required” attribute, and without a “multiple” attribute, and without a “size” attribute whose value is greater than “1”, must have either an empty “value” attribute, or must have no text content. Consider either adding a placeholder option label, or adding a “size” attribute with a value equal to the number of “option” elements. (433)

Reference:
https://maxdesign.com.au/articles/select-required/

**Comment by woodruffw** [comment]
I'm having a bit of trouble identifying this one: I see one `<select>` in that template:

```html
<select class="change-field" name="role_name" data-original="{{ role.role_name }}">
```

But it doesn't have a `required` attribute or another of the other attributes that would constitute the error, unless I'm missing something.

**Comment by nlhkabu** [comment]
@woodruffw I think it's here:
https://github.com/pypa/warehouse/blob/master/warehouse/templates/manage/roles.html#L138

The `<select>` isn't present in the template, it's being pulled in by `{{ form.role_name }}`. Therefore the required attribute must be defined in the backend.

If I visit `http://localhost/manage/project/pip/collaboration/`, I get the following HTML when inspecting the 'role' select on the form where users can add a collaborator:

![Screenshot from 2019-08-13 19-23-18](https://user-images.githubusercontent.com/3323703/62967160-52cdb080-be00-11e9-86e9-f0d5d6d32514.png)

![Screenshot from 2019-08-13 19-23-39](https://user-images.githubusercontent.com/3323703/62967159-52351a00-be00-11e9-8c34-c20326dc32c3.png)

**Comment by woodruffw** [comment]
Hmm, this might actually be a problem with WTForms' HTML generation. Here's the underlying form mixin:

```python
class RoleNameMixin:

    role_name = wtforms.SelectField(
        "Select role",
        choices=[("Maintainer", "Maintainer"), ("Owner", "Owner")],
        validators=[wtforms.validators.DataRequired(message="Select role")],
    )
```

So the fix here is probably upstream -- `choices` (or the underlying generator that uses it) needs to properly set the `size` attribute. Alternatively, we could add an additional `("", "Select role")` choice, although I'm not sure what that does at the validation level. I'll try it out.
